opentelemetry/proto/collector/logs/v1/logs_service.proto

```diff
@@ -42,6 +42,17 @@ message ExportLogsServiceRequest {
   // data from multiple origins typically batch the data before forwarding further and
   // in that case this array will contain multiple elements.
   repeated opentelemetry.proto.logs.v1.ResourceLogs resource_logs = 1;
+
+  // An idempotency key is optional information used in a server to
+  // detect retried requests and avoid double-counting of logging
+  // events.
+  //
+  // OTLP clients SHOULD include a randomly generated idempotency key.
+  // Values of size less than 8 bytes are invalid and MUST be ignored.
+  //
+  // [V4 UUID]([RFC 4122](https://tools.ietf.org/html/rfc4122)) is the
+  // recommended idempotency key format.
+  string idempotency_key = 2;
```

Comment:
I see you changed it from bytes to string. This is probably OK, but it is bigger on the wire. We could keep the bytes as well, so that it is typically 8 bytes of UUID on the wire instead of 36 bytes in string representation.

If it is a string then this comment probably needs to be re-worded since it refers to bytes and it is not clear what that means:

>Values of size less than 8 bytes are invalid and MUST be ignored.

My slight preference is to keep it as bytes and recommend V4 UUID in the bytes.

Reply:
OK. I have seen 8-byte keys in use, so I was hesitating to require a 16-byte key. I looked at the benchmarks in https://github.com/satori/go.uuid which suggest a UUIDv4 takes less than 1 microsecond to generate, which seems like not a big deal. So I've reworded this (in the three files) to say:

```
  // OTLP clients SHOULD include an idempotency key.  Empty
  // idempotency keys are invalid and MUST be ignored.  A 128-bit [V4
  // UUID]([RFC 4122](https://tools.ietf.org/html/rfc4122)) is the
  // recommended idempotency key format.
  bytes idempotency_key = 2;
```

Reply:
I am fine with if you want to only _recommend_ 16 byte UUIDs, but not require it. My concern was primarily around using bytes vs string as the data type.